graphein/grn/parse_regnetwork.py

```diff
@@ -0,0 +1,152 @@
+# %%
+# Graphein
+# Author: Ramon Vinas, Arian Jamasb <arian@jamasb.io>
+# License: MIT
+# Project Website: https://github.com/a-r-j/graphein
+# Code Repository: https://github.com/a-r-j/graphein
+import logging
+import os
+import zipfile
+from pathlib import Path
+from typing import Any, Dict, List, Union
+
+import pandas as pd
+import wget
+
+log = logging.getLogger(__name__)
+
+
+def _download_RegNetwork():
+    """
+    Downloads RegNetwork regulatory interactions
+    """
+    url = "http://www.regnetworkweb.org/download/human.zip"
+
+    # TODO: perhaps this would be better?
+    #  https://stackoverflow.com/questions/25389095/python-get-path-of-root-project-structure
+    root_dir = os.path.dirname(os.path.realpath("../"))
+    regnetwork_dir = "{}/datasets/regnetwork".format(root_dir)
+    Path(regnetwork_dir).mkdir(parents=False, exist_ok=True)
+    compressed_file = "{}/human.zip".format(regnetwork_dir)
+    out_dir = "{}/human".format(regnetwork_dir)
+    file = "{}/human.source".format(out_dir)
+
+    # Download data and unzip
+    if not os.path.exists(file):
+        print("Downloading RegNetwork ...")
+        wget.download(url, compressed_file)
+
+        with zipfile.ZipFile(compressed_file, "r") as zip_ref:
+            zip_ref.extractall(out_dir)
+
+    return file
+
+
+def _download_RegNetwork_regtypes():
+    """
+    Downloads RegNetwork regulatory interactions types
+    """
+    url = "http://www.regnetworkweb.org/download/RegulatoryDirections.zip"
+
+    root_dir = os.path.dirname(os.path.realpath("../"))
+    regnetwork_dir = "{}/datasets/regnetwork".format(root_dir)
+    Path(regnetwork_dir).mkdir(parents=False, exist_ok=True)
+    compressed_file = "{}/RegulatoryDirections.zip".format(regnetwork_dir)
+    out_dir = "{}/human".format(regnetwork_dir)
+    file = "{}/new_kegg.human.reg.direction.txt".format(out_dir)
+
+    # Download data and unzip
+    if not os.path.exists(file):
+        print("Downloading RegNetwork reg types ...")
+        wget.download(url, compressed_file)
+
+        with zipfile.ZipFile(compressed_file, "r") as zip_ref:
+            zip_ref.extractall(out_dir)
+
+    return file
+
+
+def parse_RegNetwork(gene_list: List[str], **kwargs) -> pd.DataFrame:
+    """
+    Parser for RegNetwork interactions
+    :param gene_list: List of gene identifiers
+    :return Pandas dataframe with the regulatory interactions between genes in the gene list
+    """
+    file = _download_RegNetwork()
+    df = pd.read_csv(
```

Comment:
Would it be worth having a separate loading function for RegNetwork that we can cache with `functools.lru_cache`. Would maybe give a speed up if users are making several graphs from RegNetwork

Reply:
Also, same for TRRUST